### PR TITLE
shared/storage/simple: just a simple read-only TLS store

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: shared/x509utils/names.go [hunk 1]
```diff
@@ -91,3 +91,22 @@ func removeZone(name string) string {
...
+	if addr, err := core.ParseAddr(name); err != nil {
+		s := fmt.Sprintf("[%s]", addr)
+		return s, true
+	}
+	return "", false
+}
+
+// NameAsSuffix prepares a santised hostname for matching
+// certificate patterns
+func NameAsSuffix(name string) (string, bool) {
```
Comment (karasz) [inline shared/x509utils/names.go:106]: This whole function looks a bit fishy to me.

**Review by karasz — CHANGES_REQUESTED** [review]

File: shared/x509utils/names.go [hunk 2]
```diff
@@ -91,3 +91,22 @@ func removeZone(name string) string {
 	}
 	return name[:idx]
 }
+
+// NameAsIP prepares a sanitised IP address name for matching certificates
+func NameAsIP(name string) (string, bool) {
+	if addr, err := core.ParseAddr(name); err != nil {
```
[the remark this answers is not in the document]
Comment (amery) [reply shared/x509utils/names.go]: fixed

**Comment by amery** [comment]
rebased, fixed, tests added, and `make test` made non-verbose

File: shared/x509utils/names.go (hunk 1 above)
Comment (amery) [reply shared/x509utils/names.go:106]: I thought about using a list of TLDs for safety, but this is to match against an already sanitised store so it should be harmless @karasz

**Comment by amery** [comment]
PoolBuffer.Certificates() also returns keyless certs which aren't of interest to this Store

**Review by karasz — APPROVED** [review]
LGTM except some spelling of "sanitized"

**Comment by amery** [comment]
I'm withdrawing this PR until I find the cause of a crash @karasz

**Comment by amery** [comment]
> LGTM except some spelling of "sanitized"

I believe that's `en_GB` vs `en_US` but I might be wrong

**Pull request closed without merging**